File: src/main/kotlin/me/sd_master92/customvoting/constants/enumerations/VotePartyType.java
package me.sd_master92.customvoting.constants.enumerations;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.Settings;

import java.util.Arrays;
import java.util.Optional;

public enum VotePartyType
{
    RANDOM_CHEST_AT_A_TIME(0, "Randomly"),
    ALL_CHESTS_AT_ONCE(1, "All Chests at Once"),
    ONE_CHEST_AT_A_TIME(2, "One Chest at a Time"),
    ADD_TO_INVENTORY(3, "Add To Inventory");

    private final int value;
    private final String name;

    VotePartyType(final int dropType, final String name)
    {
        this.value = dropType;
        this.name = name;
    }

    public static VotePartyType next(Main plugin)
    {
        int currentValue = VotePartyType.valueOf(plugin.getConfig().getNumber(Settings.VOTE_PARTY_TYPE)).getValue();
        if (currentValue < values().length - 1)
        {
            return valueOf(currentValue + 1);
        } else
        {
            return valueOf(0);
        }
    }

    public static VotePartyType valueOf(int value)
    {
        Optional<VotePartyType> votePartyType = Arrays.stream(values())
                .filter(type -> type.value == value)
                .findFirst();
        return votePartyType.orElse(ALL_CHESTS_AT_ONCE);
    }

    public int getValue()
    {
        return value;
    }

    public String getName()
    {
        return name;
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/constants/enumerations/VotePartyType.kt
package me.sd_master92.customvoting.constants.enumerations

import me.sd_master92.customvoting.constants.interfaces.CarouselEnum
import me.sd_master92.customvoting.constants.interfaces.EnumCompanion
import java.util.*

enum class VotePartyType(private val label_: PMessage) : CarouselEnum
{
    RANDOMLY(PMessage.ENUM_VOTE_PARTY_TYPE_RANDOM),
    RANDOM_CHEST_AT_A_TIME(PMessage.ENUM_VOTE_PARTY_TYPE_RANDOM_CHEST_AT_A_TIME),
    ALL_CHESTS_AT_ONCE(PMessage.ENUM_VOTE_PARTY_TYPE_ALL_CHESTS_AT_ONCE),
    ONE_CHEST_AT_A_TIME(PMessage.ENUM_VOTE_PARTY_TYPE_ONE_CHEST_AT_A_TIME),
    ADD_TO_INVENTORY(PMessage.ENUM_VOTE_PARTY_TYPE_ADD_TO_INVENTORY),
    EXPLODE_CHESTS(PMessage.ENUM_VOTE_PARTY_TYPE_EXPLODE_CHESTS),
    SCARY(PMessage.ENUM_VOTE_PARTY_TYPE_SCARY),
    PIG_HUNT(PMessage.ENUM_VOTE_PARTY_TYPE_PIG_HUNT),
    LOCKED_CRATES(PMessage.ENUM_VOTE_PARTY_TYPE_LOCKED_CRATES);

    fun label(locale: Locale? = null): String
    {
        return label_.getValue(locale)
    }

    override fun next(): VotePartyType
    {
        return if (ordinal < entries.size - 1)
        {
            valueOf(ordinal + 1)
        } else
        {
            valueOf(0)
        }
    }

    companion object : EnumCompanion
    {
        fun random(): VotePartyType
        {
            return entries[Random().nextInt(1, entries.size)]
        }

        override fun valueOf(key: Int): VotePartyType
        {
            return try
            {
                entries[key]
            } catch (_: Exception)
            {
                entries[0]
            }
        }
    }
}